spec/controllers/students_controller_spec.rb

```diff
@@ -0,0 +1,96 @@
+require 'rails_helper'
+
+RSpec.describe StudentsController, type: :controller do
+    #save instance of student in test db 
+    let!(:student) { create(:student, title: "Mr", first_name: "Harry", middle_name: "James", last_name: "Potter", email: "harry@test.com", gender: "Male", birth_date: "1989-07-23") }
+    
+    describe "GET #index" do
+        subject { get :index }
+
+        it "has a 200 status code" do
+            subject
+            expect(subject.status).to eq(200)
```

Comment:
@Luke-kb  here your actually calling subject twice.
If you removed line 11 totally, the result should be still the same
as line 12 will call `subject` and then status on that subject.
Try that .. 
However there is a more common way that you will see the subject pattern being used.
That is with a One-liner syntax
https://relishapp.com/rspec/rspec-core/docs/subject/one-liner-syntax

I suggest you see if you can refactor the other specs below to the oneliner syntax
```
    subject { get :index }
    
    it { is_expected.to be_success }
    it { is_expected.to render_template("students/index") }
    it { expect(assigns(:students)).to eq([student]) }

```